cabby/rvs/speak.py

```diff
@@ -16,14 +16,64 @@
 
 from typing import Text
 
-def describe_route(pivot_poi: Text, goal_poi: Text) -> Text:
+from cabby.geo import directions
+
+# Use alias for Direction to keep code cleaner.
+Direction = directions.Direction
+
+def describe_meeting_point(
+  pivot: Text, goal: Text,
+  bearing_pivot_goal: float, distance_pivot_goal: float) -> Text:
   '''Preliminary example template for generating an RVS instruction.
   
   Arguments:
-    pivot_poi: The POI used to orient with respect to the goal.
-    goal_poi: The POI that is the intended meeting location.
+    pivot: The POI used to orient with respect to the goal.
+    goal: The POI that is the intended meeting location.
+    bearing_pivot_goal: The relative bearing of the goal when heading to the
+      pivot from the start point.
+    distance_pivot_goal: The distance from the pivot to the goal.
   Returns:
     A string describing the goal location with respect to the reference.
   
   '''
-  return f'go to the {goal_poi} near the {pivot_poi}.'
\ No newline at end of file
+  direction_description = describe_egocentric_direction(
+      directions.get_egocentric_direction(bearing_pivot_goal), 
+      distance_pivot_goal)
+
+  return f'go to {goal}, which is {direction_description} {pivot}'
+
+def describe_distance(distance_in_km: float) -> Text:
+  """Convert a raw distance into a short verbal description."""
+  
+  # Round to 100 meters.
+  rdist = round(distance_in_km, 1)
+
+  if rdist < 0.1:
+    return 'not even 100 meters'
+  if rdist < 1.0:
+    return f'{int(rdist*1000)} meters'
+  else:
+    return f'{rdist}km'
+
+
+def describe_egocentric_direction(
+  direction: Direction, distance: float) -> Text:
+  """Convert a Direction and distance into a verbal description.
+  
+  This is very rough cut, to be improved considerably.
```

Comment:
I wanted to make the instructions relative to egocentric (relative) perspective rather than allocentric (absolute). There are definitely problems with the current stub solution such as obstacles, but I just wanted to get the general shape of the templating in place. Can we leave this as is and refine in future?

For me 'Y is left of X' definitely means "if you take a left from X, you will reach Y" not the other way around. Or did you mean something else?